turbinia/lib/recipe_helpers.py

```diff
@@ -0,0 +1,166 @@
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Library to contain recipe validation logic."""
+
+import copy
+import logging
+import yaml
+from yaml import Loader
+from yaml import load
+from turbinia.lib.file_helpers import file_to_str
+from turbinia.lib.file_helpers import file_to_list
+from turbinia.client import TASK_MAP
+
+log = logging.getLogger('turbinia')
+
+#Attributes allowed on the 'globals' task recipe
+DEFAULT_GLOBALS_RECIPE = {
+    'debug_tasks': False,
+    'jobs_allowlist': [],
+    'jobs_denylist': [],
+    'yara_rules_file': None,
+    'filter_patterns_file': None,
+    'yara_rules': '',
+    'filter_patterns': ''
+}
+
+#Default recipes dict
+DEFAULT_RECIPE = {'globals': DEFAULT_GLOBALS_RECIPE}
+
+
+def load_recipe_from_file(recipe_file):
+  """Load recipe from file.
+
+  Args:
+    recipe_file(str): Name of the recipe file to be read.
+
+  Returns:
+    dict: Validated and corrected recipe dictionary.
+        Empty dict if recipe is invalid.
+  """
+  if not recipe_file:
+    return copy.deepcopy(DEFAULT_RECIPE)
+  try:
+    with open(recipe_file, 'r') as r_file:
+      recipe_file_contents = r_file.read()
+      recipe_dict = load(recipe_file_contents, Loader=Loader)
+      if validate_recipe(recipe_dict):
+        return recipe_dict
+  except yaml.parser.ParserError as exception:
+    message = (
+        'Invalid YAML on recipe file {0:s}: {1!s}.'.format(
+            recipe_file, exception))
+    log.error(message)
+  except IOError as exception:
+    log.error(
+        'Failed to read recipe file {0:s}: {1!s}'.format(
+            recipe_file, exception))
+  return {}
+
+
+def validate_globals_recipe(proposed_globals_recipe):
+  """Validate the 'globals' special task recipe.
+
+  Args:
+    proposed_globals_recipe(dict): globals task recipe in need of validation.
+
+  Returns:
+    Bool indicating whether the recipe has a valid format.
+  """
+  reference_globals_recipe = copy.deepcopy(DEFAULT_GLOBALS_RECIPE)
+  reference_globals_recipe.update(proposed_globals_recipe)
+
+  filter_patterns_file = proposed_globals_recipe.get(
+      'filter_patterns_file', None)
+  yara_rules_file = proposed_globals_recipe.get('yara_rules_file', None)
+  if filter_patterns_file:
+    proposed_globals_recipe['filter_patterns'] = file_to_list(
+        filter_patterns_file)
+  if yara_rules_file:
+    proposed_globals_recipe['yara_rules'] = file_to_str(yara_rules_file)
+  diff = set(proposed_globals_recipe) - set(DEFAULT_GLOBALS_RECIPE)
+  if diff:
+    log.error(
+        'Unknown keys [{0:s}] found on globals recipe item'.format(str(diff)))
+    return False
+
+  if ('jobs_allowlist' in proposed_globals_recipe and
+      'jobs_denylist' in proposed_globals_recipe):
```

Comment:
I just realized that this won't allow the recipe to even have empty values for these keys, but what we probably want to check for is whether there is actually values set by these.   You might change this to the following (sorry for not catching that before):

```suggestion
  if (proposed_globals_recipe.get('jobs_allowlist') and
      proposed_globals_recipe.get( 'jobs_denylist')):
```